minclient/connection.go

```diff
@@ -717,6 +715,7 @@ func (c *connection) getConsensus(ctx context.Context, epoch uint64) (*commands.
 		c.Unlock()
 		return nil, ErrNotConnected
 	}
+	c.Unlock()
```

Comment:
since we're repeating this, wonder if it would make sense to have a method like
```go
func (c *connection) isConnectedMethod() (err) {
  c.Lock()
  if !c.isConnected {
    err = ErrNotConnected
  }
  c.Unlock()
}
```

Reply:
I dont feel that results in signifcant deduplication, while adding a small overhead